crates/analytics/src/lib.rs

```diff
@@ -0,0 +1,229 @@
+use async_trait::async_trait;
+use std::ops::Div;
+
+use rain_orderbook_subgraph_client::types::common::Trade;
+use rain_orderbook_subgraph_client::{OrderbookSubgraphClient, OrderbookSubgraphClientError};
+
+#[async_trait]
+pub trait OrderbookSubgraphClientTrait {
```

Comment:
I'm creating this for two reasons:

1. To make the code more testable
2. To not pollute the `OrderbookSubgraphClient` with analytics-related methods